packages/react-resize-handle/src/hooks/useUnitHandle.ts

```diff
@@ -0,0 +1,115 @@
+import { useFluent } from '@fluentui/react-components';
+import * as React from 'react';
+
+import type { CSSUnit, CSSUnitName, GrowDirection } from '../types';
+
+export type UnitHandle = {
+  elementDimension: (element: HTMLElement | null) => number;
+  fromPxToValue: (px: number) => number;
+  getOffsetStep: () => number;
+  roundValue: (value: number) => number;
+  name: CSSUnitName;
+};
+
+const DEFAULT_OFFSET_STEP = 20;
+
+export function useUnitHandle(
+  growDirection: GrowDirection,
+  unit: CSSUnit
+): UnitHandle {
+  const { targetDocument } = useFluent();
+
+  const windowDimensionsRef = React.useRef<{ width: number; height: number }>({
+    width: 0,
+    height: 0,
+  });
+  const windowDimensionsRafId = React.useRef<number | null>(null);
+
+  React.useEffect(() => {
+    const targetWindow = targetDocument?.defaultView;
+
+    if (targetWindow && unit === 'viewport') {
+      windowDimensionsRef.current.width = targetWindow.innerWidth;
+      windowDimensionsRef.current.height = targetWindow.innerHeight;
+
+      const handleResize = () => {
+        windowDimensionsRafId.current = targetWindow.requestAnimationFrame(
+          () => {
+            windowDimensionsRef.current.width = targetWindow.innerWidth;
+            windowDimensionsRef.current.height = targetWindow.innerHeight;
+          }
+        );
+      };
+
+      targetWindow.addEventListener('resize', handleResize);
+
+      return () => {
+        targetWindow.removeEventListener('resize', handleResize);
+
+        if (windowDimensionsRafId.current) {
+          targetWindow.cancelAnimationFrame(windowDimensionsRafId.current);
+          windowDimensionsRafId.current = null;
+        }
+      };
+    }
+  }, [targetDocument, unit]);
+
+  return React.useMemo(() => {
+    const name: CSSUnitName =
```

Comment:
instead of if branches in the handle, can we just have separate implementations based on the selected unit?